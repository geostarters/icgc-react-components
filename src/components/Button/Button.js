//@flow

import React from "react";
import PropTypes from "prop-types";
import { Button as ButtonSUI } from "semantic-ui-react";

export const Button = props => (<ButtonSUI {...props}></ButtonSUI>);

export default Button;

Button.propTypes = {
	color: PropTypes.string,
	content: PropTypes.string,
	onClick: PropTypes.func,
};

Button.defaultProps = {
	color: "yellow",
	content: "",
	onClick: () => console.log("click")
};
