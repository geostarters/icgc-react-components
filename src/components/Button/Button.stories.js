//@flow

import React from "react";

import Button from "./Button";

import "semantic-ui-css/semantic.min.css";

export default {
	title: "Button",
	component: Button
};

const Template = args => (<Button {...args}/>);

export const OkButton = Template.bind({});
OkButton.storyName = "Ok button";
OkButton.args = {
	content:"OK"
};

export const IconButton = Template.bind({});
IconButton.storyName = "Icon button";
IconButton.args = {
	icon:"sign-in"
};

export const BasicButton = Template.bind({});
BasicButton.storyName = "Basic button";
BasicButton.args = {
	basic: true,
	content: "Basic"
};

export const CancelButton = Template.bind({});
CancelButton.storyName = "Cancel button";
CancelButton.args = {
	color: "red",
	content: "Cancel"
};
