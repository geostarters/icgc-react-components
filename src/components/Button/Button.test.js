//@flow

import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import React from "react";
import Button from "./Button";

describe("Button component", () => {

	const mockProps = {
		color: "teal",
		content: "OK",
		onClick: jest.fn()
	};
	const component = shallow(<Button {...mockProps}></Button>);

	it("to match the snapshot ", () => {

		expect(toJSON(component)).toMatchSnapshot();

	});

	it("has color prop = teal ", () => {

		expect(component.props().color).toEqual("teal");

	});

	it("has a prop function called on click", () => {

		component.simulate("click");
		expect(mockProps.onClick).toBeCalled();

	});


});
