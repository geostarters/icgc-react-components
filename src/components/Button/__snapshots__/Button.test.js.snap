// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component to match the snapshot  1`] = `
<Button
  as="button"
  color="teal"
  onClick={[Function]}
>
  OK
</Button>
`;
